src/fragments/lib/auth/common/totp/flows.mdx

```diff
@@ -0,0 +1,89 @@
+
+## Setup
+
+<Callout warning>
+
+The following options are only available when starting a new project (via `amplify add auth`). You will not have access to these settings after creation (via `amplify update`).
+
+- Required MFA
+
+</Callout>
+
+<BlockSwitcher>
+
+<Block name="New Project">
+
+Run `amplify add auth` to create a new Cognito Auth resource, and follow the prompts below to set up TOTP in your flow.
+
+
+### As a verification method
+
+import all1 from "/src/fragments/lib/auth/common/totp/add_verification.mdx";
+
+<Fragments fragments={{all: all1}} />
+
+### TOTP MFA
+
+import all2 from "/src/fragments/lib/auth/common/totp/add_mfa.mdx";
+
+<Fragments fragments={{all: all2}} />
+
+</Block>
+
+<Block name="Existing Auth">
+
+Run `amplify update auth` and follow the prompts as guided below.
+
+### As a verification method
+
+import all4 from "/src/fragments/lib/auth/common/totp/update_verification.mdx";
+
+<Fragments fragments={{all: all4}} />
+
+### TOTP MFA
+
+import all5 from "/src/fragments/lib/auth/common/totp/update_mfa.mdx";
+
+<Fragments fragments={{all: all5}} />
+
+</Block>
+
+</BlockSwitcher>
+
+## Sign In
+
+When MFA is marked as Required or **ON** for the user, and TOTP has been not been set up, a `signIn` call will return `continueSignInWithTOTPSetup` as the next step. This step signifies that the user must set up TOTP before they can sign in. The step returns an associated value of type `TOTPSetupDetails` that contains the `sharedSecret` that will be used to either to generate a QR code or can be manually entered into an Authenticator app. Once the authenticator app is set up, the user can generate a TOTP code and provide it to the library to complete the sign in process.
+
+import iosSignIn from "/src/fragments/lib/auth/ios/totp/sign_in.mdx";
+
+<Fragments fragments={{ios: iosSignIn}} />
+
+The TOTP code can be obtained from the user via a text field or any other means. Once the user provides the TOTP code, call `confirmSignIn` with the TOTP code as the `challengeResponse` parameter.
+
+Note: This process of setting up MFA is only valid for TOTP.
+
+`TOTPSetupDetails` also provides a helper method called `getSetupURI` that can be used to generate a URI, which can be used by native password managers for TOTP association. If the URI is used on Apple devices, it will trigger the platform's native password manager to associate the TOTP code with the account.
+
+import iosConfirmSignIn from "/src/fragments/lib/auth/ios/totp/confirm_sign_in.mdx";
+
+<Fragments fragments={{ios: iosConfirmSignIn}} />
+
+## TOTP Setup After Sign In
+
+If MFA is marked as **Optional** in Cognito User Pool, and TOTP is marked as an enabled MFA method for the user, then the user can set up TOTP after sign in. This can be achieved by calling `setUpTOTP` and `verifyTOTPSetup` APIs that are available in the `Auth` category.
```

Comment:
```suggestion
If MFA is marked as **Optional** in Cognito User Pool, and TOTP is marked as an enabled MFA method for the user, then the user can set up TOTP after sign in. This can be achieved by calling the `setUpTOTP` and `verifyTOTPSetup` APIs that are available in the `Auth` category.
```

Reply:
> and TOTP is marked as an enabled MFA method for the user

Isn't this true by default?

Reply:
I guess CLI you it enabled by default and in the web app you have to explicitly select TOTP.